src/envoy/admin/manager/site_reading.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from datetime import datetime
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from envoy.admin.crud.site_reading import (
+    count_site_readings_for_site_and_time,
+    select_agg_id_for_site,
+    select_csip_aus_site_type_ids,
+    select_site_readings_for_site_and_time,
+)
+from envoy.admin.mapper.site_reading import AdminSiteReadingMapper
+from envoy_schema.admin.schema.site_reading import CSIPAusSiteReadingUnit, CSIPAusSiteReadingPageResponse
+
+
+class AdminSiteReadingManager:
+    """Logic layer for admin site reading operations."""
+
+    @staticmethod
+    async def get_site_readings_for_site_and_time(
+        session: AsyncSession,
+        site_id: int,
+        csip_unit: CSIPAusSiteReadingUnit,
+        start_time: datetime,
+        end_time: datetime,
+        start: int = 0,
+        limit: int = 500,
+    ) -> CSIPAusSiteReadingPageResponse:
+        """Get site readings for specified site within a time range."""
+
+        # Convert CSIP unit to UOM for database query
+        uom = AdminSiteReadingMapper.csip_unit_to_uom(csip_unit)
+
+        # Query 1: get the aggregator id from site to help hit the index of SiteReadingType
+        aggregator_ids = await select_agg_id_for_site(session=session, site_id=site_id)
+
+        # Query 2: retrieve the reading type IDs for this site in order to hit index of SiteReading
+        site_type_ids = await select_csip_aus_site_type_ids(
+            session=session, aggregator_ids=aggregator_ids, site_id=site_id, uom=uom
+        )
+
+        # If no aggregator_ids or site_type_ids found, return empty result i
+        if not aggregator_ids or not site_type_ids:
+            return AdminSiteReadingMapper.map_to_csip_aus_reading_page_response(
+                site_readings=[],
+                total_count=0,
+                start=start,
+                limit=limit,
+                site_id=site_id,
+                start_time=start_time,
+                end_time=end_time,
+                requested_unit=csip_unit,
+            )
+
+        # Queries 3: Get total count and readings in parallel
+        total_count_task = count_site_readings_for_site_and_time(
+            session=session,
+            site_type_ids=site_type_ids,
+            start_time=start_time,
+            end_time=end_time,
+        )
+
+        site_readings_task = select_site_readings_for_site_and_time(
+            session=session,
+            site_type_ids=site_type_ids,
+            start_time=start_time,
+            end_time=end_time,
+            start=start,
+            limit=limit,
+        )
+
+        total_count, site_readings = await asyncio.gather(total_count_task, site_readings_task)
```

Comment:
This is fine but I figured I'd add a word of warning about trying to execute these requests simultaneously (rather than just awaiting them sequentially). 

Both of these requests are "read only" so you're safe - but be warned that raised exceptions will NOT cancel subsequent requests.